src/test/java/org/skife/jdbi/v2/sqlobject/BindInNullTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.skife.jdbi.v2.sqlobject;
+
+import org.h2.jdbcx.JdbcDataSource;
+import org.hamcrest.CoreMatchers;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.Something;
+import org.skife.jdbi.v2.logging.FormattedLog;
+import org.skife.jdbi.v2.sqlobject.stringtemplate.UseStringTemplate3StatementLocator;
+import org.skife.jdbi.v2.tweak.SQLLog;
+import org.skife.jdbi.v2.unstable.BindIn;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.skife.jdbi.v2.unstable.BindIn.EmptyHandling.NULL;
+import static org.skife.jdbi.v2.unstable.BindIn.EmptyHandling.VOID;
+
+public class BindInNullTest
+{
+    private static Handle handle;
+
+    @BeforeClass
+    public static void init()
+    {
+        final JdbcDataSource ds = new JdbcDataSource();
+        ds.setURL("jdbc:h2:mem:" + UUID.randomUUID());
+        final DBI dbi = new DBI(ds);
+        dbi.registerMapper(new SomethingMapper());
+        handle = dbi.open();
+
+        handle.execute("create table something (id int primary key, name varchar(100))");
+        handle.execute("insert into something(id, name) values(1, null)");
+        handle.execute("insert into something(id, name) values(2, null)");
+        handle.execute("insert into something(id, name) values(3, null)");
+        handle.execute("insert into something(id, name) values(4, null)");
+
+        // "control group" elements that should *not* be returned by the queries
+        handle.execute("insert into something(id, name) values(5, 'bla')");
+        handle.execute("insert into something(id, name) values(6, 'null')");
+        handle.execute("insert into something(id, name) values(7, '')");
+    }
+
+    @AfterClass
+    public static void exit()
+    {
+        handle.execute("drop table something");
+        handle.close();
+    }
+
+    @Test
+    public void testSomethingByIterableHandleNullWithNull()
+    {
+        final SomethingByIterableHandleNull s = handle.attach(SomethingByIterableHandleNull.class);
+
+        final List<Something> out = s.get(null);
```

Comment:
This might be considered a programming error -- should we really assume `null` is `empty list` instead of a coding error?